### PR TITLE
[Tracing] Refactor DD_TAGS test cases for clarity

### PR DESCRIPTION
**Pull request by zacharycmontoya**
## Motivation

The tracing libraries are expected to parse `DD_TAGS` and assign them as span tags. Since this is both an agent setting and a tracer setting, the parsing logic is not trivial.

## Changes

This PR refactors the test cases to indicate what the single `DD_TAGS` environment variable is and the resulting pairs of tags set on the spans.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/parametric/test_config_consistency.py [hunk 1]
```diff
@@ -281,30 +281,36 @@ def test_setting_trace_rate_limit(self, library_env, test_agent, test_library):
...
-    env10: dict = {"DD_TAGS": "a:b,c,d"}
-    env11: dict = {"DD_TAGS": "a,1"}
-    env12: dict = {"DD_TAGS": "a:b:c:d"}
-    return parametrize("library_env", [env1, env2, env3, env4, env5, env6, env7, env8, env9, env10, env11, env12])
+tag_scenarios: dict = {
+    "key1:value1,key2:value2": [("key1", "value1"), ("key2", "value2")],
+    "key1:value1 key2:value2": [("key1", "value1"), ("key2", "value2")],
+    "env:test aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal"), ("bKey", "bVal"), ("cKey", "")],
+    "env:test,aKey:aVal,bKey:bVal,cKey:": [("env", "test"), ("aKey", "aVal"), ("bKey", "bVal"), ("cKey", "")],
+    "env:test,aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal bKey:bVal cKey:")],
```
Comment (bm1549) [inline tests/parametric/test_config_consistency.py:289]: Should we also add some test scenarios for keys/values as numbers, booleans, special characters, and other edge cases?

```suggestion
    "env:test,aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal bKey:bVal cKey:")],
    "numKey:1,1:numVal,boolKey:true,true:boolVal": [("numKey", "1"), ("1", "numVal"), ("boolKey", "true"), ("true", "boolVal") ],
    "": [],
    "sp%key:value,key:sp#value": [], # I'm not sure what belongs here
```

Comment (zacharycmontoya) [reply tests/parametric/test_config_consistency.py:289]: Yes, but I would prefer to do this in a separate PR. Some of the existing test cases are confusing to me, so I want to revisit the existing test cases and add to them as we build consensus

Comment (bm1549) [reply tests/parametric/test_config_consistency.py:289]: SGTM

**Review by bm1549 — APPROVED** [review]
